core/src/test/java/bio/terra/pearl/core/service/export/EnrolleeImportServiceTests.java

```diff
@@ -73,23 +74,73 @@ public class EnrolleeImportServiceTests extends BaseSpringBootTest {
+        ParticipantUser user2 = participantUserService.find(dataImport.getImportItems().get(1).getCreatedParticipantUserId()).orElseThrow();
+        Enrollee enrollee2 = enrolleeService.findByParticipantUserIdAndStudyEnvId(user2.getId(), bundle.getStudyEnv().getId()).orElseThrow();
+
+        //now try update
+        Import dataImportUpd = importAndVerify(bundle, savedAdmin, csvStringUpdate, "1982-10-10");
+        ImportItem importItem = dataImportUpd.getImportItems().get(0);
+        ParticipantUser userUpdated = participantUserService.find(importItem.getCreatedParticipantUserId()).orElseThrow();
+        Enrollee enrolleeUpdated = enrolleeService.findByParticipantUserIdAndStudyEnvId(userUpdated.getId(), bundle.getStudyEnv().getId()).orElseThrow();
+        //should be the same participant and enrollee and profile
+        assertThat(userUpdated.getId(), equalTo(user.getId()));
+        assertThat(enrollee.getId(), equalTo(enrolleeUpdated.getId()));
+        assertThat(enrollee.getProfileId(), equalTo(enrolleeUpdated.getProfileId()));
+        //load profile and verify that birthDate was updated
+        Profile profileUpdated = profileService.find(enrolleeUpdated.getProfileId()).orElseThrow();
+        assertThat(profileUpdated.getBirthDate(), equalTo(LocalDate.parse("1982-10-10")));
+        //load and verify that new birthDate was inserted for user2
+        profileUpdated = profileService.find(enrollee2.getProfileId()).orElseThrow();
+        assertThat(profileUpdated.getBirthDate(), equalTo(LocalDate.parse("1990-10-10")));
+
+        //same user different portal.. should create new profile
+        String csvStringPortal2 = """
+                account.username,account.createdAt,enrollee.createdAt,profile.birthDate
+                userName1,"2024-05-09 01:37PM","2024-05-09 01:38PM","1990-10-10"
+                userName2,"2024-05-11 10:00AM","2024-05-11 10:00AM"
+                """;
+        //same participant, diff profile
+        Import dataImportUpd2 = importAndVerify(bundle2, savedAdmin, csvStringPortal2, "1990-10-10");
+        ImportItem importItem2 = dataImportUpd2.getImportItems().get(0);
+        ParticipantUser userUpdated2 = participantUserService.find(importItem2.getCreatedParticipantUserId()).orElseThrow();
+        Enrollee enrolleeUpdated2 = enrolleeService.findByParticipantUserIdAndStudyEnvId(userUpdated2.getId(), bundle2.getStudyEnv().getId()).orElseThrow();
+        //should be the same participant and enrollee and profile
+        assertThat(userUpdated2.getId(), equalTo(user.getId()));
+        Assertions.assertNotEquals(enrollee.getId(), enrolleeUpdated2.getId());
+        Assertions.assertNotEquals(enrollee.getProfileId(), enrolleeUpdated2.getProfileId());
+
+    }
+
+    private Import importAndVerify(StudyEnvironmentFactory.StudyEnvironmentBundle bundle, AdminUser admin, String csvString,
```

Comment:
It took me a long time to figure out what this method did.  Consider renaming, adding a comment, and also renaming the "birthDate" parameter to clarify which birthdate

Reply:
Also, (perhaps in a later PR) consider a more generic "ImportAndVerify" method that takes as arguments an array of hamcrest Matchers -- one entry for each Enrollee so that you can more clearly specify the fields that are important

Reply:
Took care of most except "array of hamcrest Matchers" which I will look into separately.

Reply:
refactored to pass objects rather than individual attributes so that it looks good with more attributes from ParticipantUsre/Enrolle/Profile